tests/versioning.go

```diff
@@ -1448,18 +1613,26 @@ func (s *VersioningIntegSuite) dispatchUnversionedRemainsUnversioned() {
 	s.NoError(run.Get(ctx, &out))
 	s.Equal("done!", out)
 
-	s.validateWorkflowBuildId(ctx, run.GetID(), run.GetRunID(), "", false, true, "", nil)
+	s.validateWorkflowBuildId(ctx, run.GetID(), run.GetRunID(), "", false, "binary-checksum", "", nil)
+}
+
+func (s *VersioningIntegSuite) TestDispatchUpgradeStopOldOld() {
+	s.testWithMatchingBehavior(func() { s.dispatchUpgrade(false, true) })
+}
+
+func (s *VersioningIntegSuite) TestDispatchUpgradeWaitOld() {
+	s.testWithMatchingBehavior(func() { s.dispatchUpgrade(false, false) })
 }
 
 func (s *VersioningIntegSuite) TestDispatchUpgradeStopOld() {
-	s.testWithMatchingBehavior(func() { s.dispatchUpgrade(true) })
+	s.testWithMatchingBehavior(func() { s.dispatchUpgrade(true, true) })
 }
 
 func (s *VersioningIntegSuite) TestDispatchUpgradeWait() {
-	s.testWithMatchingBehavior(func() { s.dispatchUpgrade(false) })
+	s.testWithMatchingBehavior(func() { s.dispatchUpgrade(true, false) })
 }
```

Comment:
use two for loops?

Reply:
old versioning test will be cleaned up later.